### PR TITLE
Bluetooth: Host: Notify upper layer if the BR ACL is established

### PR DESCRIPTION
**Pull request by lylezhu2012**
In current implementation, the upper is not notified when BR ACL connection is established.
L2CAP does not know that ACL connection has been established. All fixed channels will not be notified. Then The SDP server cannot work properly.

### CODE REVIEW

**Review by jhedberg — CHANGES_REQUESTED** [review]
@lylezhu2012 please split the coding style changes into a separate commit, or even better a separate PR. It's very hard to see what the actual functional change is when most of the changes in the PR are something completely different.

**Comment by lylezhu2012** [comment]
> @carlescufi @jori-nordic @lylezhu2012 btw, seems like we have some tweaking to do with `MAINTAINERS.yml` to make it detect such PRs as primarily Bluetooth Classic related. Right now the assignee selection script logs show the following for this PR:
> 
> ```
> working on https://github.com/zephyrproject-rtos/zephyr/pull/67996 : Bluetooth: Host: Notify upper layer if the BR ACL is established
> file: subsys/bluetooth/host/br.c
> Area matches: {<Area Bluetooth Host>: 1, <Area Bluetooth Classic>: 1}
> labels: {'area: Bluetooth Host', 'area: Bluetooth', 'area: Bluetooth Classic'}
> collab: ['jori-nordic', 'jhedberg', 'hermabe', 'alwa-nordic', 'Thalley', 'sjanc', 'theob-pro', 'lylezhu2012']
> Submitted by: lylezhu2012
> candidate maintainers: {'jori-nordic': 1, 'jhedberg': 1, 'lylezhu2012': 1}
> Picked assignee: jori-nordic (100.00% ownership)
> +++++++++++++++++++++++++
> adding label area: Bluetooth Host...
> adding label area: Bluetooth...
> adding label area: Bluetooth Classic...
> adding reviewers ['jori-nordic', 'jhedberg', 'hermabe', 'alwa-nordic', 'Thalley', 'sjanc', 'theob-pro']...
> Adding assignee NamedUser(login="jori-nordic")...
> ```

Got it. Thanks.

**Comment by lylezhu2012** [comment]
> @lylezhu2012 please split the coding style changes into a separate commit, or even better a separate PR. It's very hard to see what the actual functional change is when most of the changes in the PR are something completely different.

Got it. Thanks.

**Review by jori-nordic — DISMISSED** [review]

**Comment by jori-nordic** [comment]
> Right now the assignee selection script logs show the following for this PR

Hm yeah I think I have to have a corresponding filter-out-list of the files listed in Bluetooth Classic for the Host maintainer section.
I wanted to put all the classic stuff in its own directory before but didn't get around to it.

@lylezhu2012 do you think you could do that? Ie. create a `subsys/bluetooth/host/classic` directory and move all the classic files there? I can make the MAINTAINERS.yml update after.

File: subsys/bluetooth/host/br.c [hunk 1]
```diff
@@ -1,4 +1,5 @@
 /*
+ * Copyright 2024 NXP
```
Comment (jhedberg) [inline subsys/bluetooth/host/br.c:2]: Could you follow the same format as for other copyright lines, i.e. "Copyright (c) <year> <company>? (you have the "(c)" missing right now.

Comment (Thalley) [reply subsys/bluetooth/host/br.c:2]: NXP usually don't add that

Comment (lylezhu2012) [reply subsys/bluetooth/host/br.c:2]: I will remove this line.

Comment (dleach02) [reply subsys/bluetooth/host/br.c]: @Thalley is correct. Our NXP legal guidance is how @lylezhu2012 is doing it. Don't ask me why.. we just do what our legal says we need to do ;-)

**Review by jhedberg — APPROVED** [review]

**Review by jori-nordic — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Comment by lylezhu2012** [comment]
> > Right now the assignee selection script logs show the following for this PR
> 
> Hm yeah I think I have to have a corresponding filter-out-list of the files listed in Bluetooth Classic for the Host maintainer section. I wanted to put all the classic stuff in its own directory before but didn't get around to it.
> 
> @lylezhu2012 do you think you could do that? Ie. create a `subsys/bluetooth/host/classic` directory and move all the classic files there? I can make the MAINTAINERS.yml update after.

It is OK for me. However, I feel that this cannot prevent non-BR/EDR from being affected when BR/EDR needs to be updated.

**Pull request merged**